Frontend/src/Components/Signup.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useContext } from 'react'
+import { Link, useNavigate } from 'react-router-dom'
+import { UserContext } from '../UserContext';
+import '../CSS/Signup.css'
+
+export default function Signup() {
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const [username, setUsername] = useState('');
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const { updateUser } = useContext(UserContext);
+    const navigate = useNavigate();
+
+    const handleSubmit = async (e) => {
+        try {
+            const response = await fetch(`http://localhost:4700/users`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({ firstName, lastName, username, email, password }),
+                credentials: 'include'
+            });
+
+            if (response.ok) {
+                const data = await response.json();
+                const loggedInUser = data.user;
+
+                console.log('Signup Successful!')
+
+                setFirstName('');
+                setLastName('');
+                setUsername('');
+                setEmail('');
+                setPassword('');
+
+                updateUser(loggedInUser)
+
+                navigate('/')
+            }else {
+                alert('Signup failed!');
+            }
+        } catch (error) {
+            alert('Signup failed' + error);
+        }
+    }
+
+    return (
+        <div className='signup-container'>
+            <form className='signup-form' onSubmit={handleSubmit}>
+                <h2>Sign Up</h2>
+                <div className='form-group'>
```

Comment:
can we add these values to a map then iterate through them? that'll keep things a bit cleaner and allow you to add more form values with ease